### PR TITLE
Update pdos.py

### PR DESCRIPTION
**Pull request by MohammadShakiba**
convolve_cp2k_pdos was added

### CODE REVIEW

File: src/libra_py/pdos.py [hunk 1]
```diff
@@ -384,4 +384,154 @@ def libra_pdos(_emin, _emax, _de, projections, prefix, outfile, Nel, do_convolve
 
     return res
     
-        
\ No newline at end of file
+
+def convolve_cp2k_pdos(cp2k_log_file: str, time_step: int, sigma: float, coef: float, npoints: int, energy_conversion: float, angular_momentum_cols: list):
```
Comment (alexvakimov) [inline src/libra_py/pdos.py:388]: there are relatively many input parameters, so could you please pack them into a dictionary. This way you can use the check_input function (please do) to setup the default values and check for critical inputs

also - do you really need the "energy_conversion" as an input parameter? it is merely a constant. for consistency, please use the "libra_py.units" module

File: src/libra_py/pdos.py [hunk 2]
```diff
@@ -384,4 +384,154 @@ def libra_pdos(_emin, _emax, _de, projections, prefix, outfile, Nel, do_convolve
...
+    
+    
+    # Now we make an equispaced energy vector from min_energy ad max_energy with npoints.
+    energy_grid = np.linspace( min_energy-2, max_energy+2, npoints )
+    energy_grid = np.array(energy_grid)
+    
+    
+    # Appending the energy lines with their component densities of states
+    energy_lines = []
+    for i in range( time_step * ( num_levels + 2 ) + 2, ( time_step + 1 ) * ( num_levels + 2 ) ):
```
Comment (alexvakimov) [inline src/libra_py/pdos.py:481]: make the inputs of the range function the pre-computed variables - don't do the calculations in the range function itself

File: src/libra_py/pdos.py [hunk 3]
```diff
@@ -384,4 +384,154 @@ def libra_pdos(_emin, _emax, _de, projections, prefix, outfile, Nel, do_convolve
...
+    
+    convolved_pdos = []
+    t1 = time.time()
+    for j in range(1,len(angular_momentum_cols)+1):
+        # Initialize a vector of zeros summing the weighted PDOS
+        tmp_weighted_pdos = np.zeros(energy_grid.shape)
+
+        for i in range(0,num_levels):
+            # The Guassian function
+            gaussian_fun = (coef/(sigma*np.sqrt(2.0*np.pi)))*(np.exp(-0.5*np.power(((energy_grid-float(pdos_sum[i][0])*energy_conversion)/sigma),2)))
```
Comment (alexvakimov) [inline src/libra_py/pdos.py:526]: perhaps, pre-compute the common pre-factor before the double loop to reduce the computational time

**Review by alexvakimov — CHANGES_REQUESTED** [review]

**Review by MohammadShakiba — COMMENTED** [review]
the input variable is now a dictionary
checking the input dictionary was added using `comn` 
some parts were fixed as mentioned in the comment
some other fixes were also applied
time library was added
the conversion value can be removed or set to 1 but the user will need to multiply the outputs (`energy_grid` and `homo_energy`) by the conversion unit they need. For the average pdos over the MD this should be applied in the `for` loop over the MD pdos files. Here, we set it as default to eV.
the function was tested successfully.

**Pull request merged**